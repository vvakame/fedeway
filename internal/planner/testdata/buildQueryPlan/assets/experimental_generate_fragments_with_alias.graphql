# schema: supergraphSdl.graphqls
# option:autoFragmentization: true

# experimental compression to downstream services
# should generate fragments correctly when aliases are used
query {
    reviews: topReviews {
        content: body
        author {
            username
        }
        product {
            name
            cost: price
            details {
                origin: country
            }
        }
    }
}
