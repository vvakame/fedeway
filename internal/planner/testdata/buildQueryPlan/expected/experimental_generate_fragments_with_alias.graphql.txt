QueryPlan {
	Sequence {
		Fetch(service: "reviews") {
			query {
				reviews: topReviews {
					... __QueryPlanFragment_1__
				}
			}
			fragment __QueryPlanFragment_1__ on Review {
				content: body
				author {
					username
				}
				product {
					... __QueryPlanFragment_0__
				}
			}
			fragment __QueryPlanFragment_0__ on Product {
				__typename
				... on Book {
					__typename
					isbn
				}
				... on Furniture {
					__typename
					upc
				}
			}
		},
		Parallel {
			Sequence {
				Flatten(path: "reviews.@.product") {
					Fetch(service: "books") {
						{
							... on Book {
								__typename
								isbn
							}
						} =>
						{
							... on Book {
								__typename
								isbn
								title
								year
							}
						}
					},
				},
				Flatten(path: "reviews.@.product") {
					Fetch(service: "product") {
						{
							... on Book {
								__typename
								isbn
								title
								year
							}
						} =>
						{
							... on Book {
								name
							}
						}
					},
				},
			},
			Flatten(path: "reviews.@.product") {
				Fetch(service: "product") {
					{
						... on Furniture {
							__typename
							upc
						}
					}
					{
						... on Book {
							__typename
							isbn
						}
					} =>
					{
						... on Furniture {
							name
							cost: price
							details {
								origin: country
							}
						}
						... on Book {
							cost: price
							details {
								origin: country
							}
						}
					}
				},
			},
		},
	},
}